### PR TITLE
I230 move sendemail next api to the frontend

### LINKED ISSUE #230: Move `sendEmail` Next API to the frontend
**Is your feature request related to a problem? Please describe.**
The current function is implemented in Next API. The function might be redundant as the function can be done on the frontend.

**Describe the solution you'd like**
Instead of using the Next API to create a `mail` document, create the `mail` docs right after the user has submitted the `vet` or `incident` form on the frontend.

Just this one line of code:
```ts
// src/pages/api/sendEmail.ts
await firestore.collection('mail').add(email)
```
Keep in mind that the above code is an API from the `firebase-admin` and it is basically just `firebase` version 8 that is using classes. Convert it to use APIs from `firebase` version 9 which uses functions.

### PR DESCRIPTION
**Pull request by NicholasChoong**
## Change Summary
- Updated doc
- email trigger is now fully working

# Related Issue

- Resolve #230

### CODE REVIEW

File: docs/functions.md [hunk 1]
```diff
@@ -20,18 +20,23 @@ Then add the extension to Firestore.
 7. Click it and search for the [Trigger Email](https://extensions.dev/extensions/firebase/firestore-send-email) extension. Click install. This should take you back to the Firebase console.
 8. Set the options to something like the following:
 
-```
+```nash
```
Comment (scratchclaggy) [inline docs/functions.md:23]: Is this a typo? Never heard of nash.

File: src/components/Visit/VetForm/index.tsx [hunk 2]
```diff
@@ -61,18 +62,11 @@ const VetForm = (props: VetFormProps) => {
...
       const message = {
         subject: 'Vet Concerns Report',
         text: formatIncident(data)
       }
-      await fetch('/api/sendEmail', {
-        method: 'POST',
-        body: JSON.stringify(message)
-      })
-
+      await Promise.all([mutateVetConcerns(data), sendEmail(message)])
```
Comment (scratchclaggy) [inline src/components/Visit/VetForm/index.tsx:69]: `mutateVetConcerns()` isn't `async`

**Review by scratchclaggy — CHANGES_REQUESTED** [review]
Haven't tested the behaviour yet, but code looks fine to me.

File: docs/functions.md (hunk 1 above)
Comment (NicholasChoong) [reply docs/functions.md:23]: def a type

**Review by scratchclaggy — APPROVED** [review]
Looks good :rocket:

**Pull request merged**